packages/ui/VirtualList/VirtualListBase.js

```diff
@@ -797,10 +771,42 @@ const VirtualListBaseFactory = (type) => {
 			delete rest.spacing;
 			delete rest.updateStatesAndBounds;
 
+			const
+				{dataSize, direction, itemSize, overhang, rtl, spacing} = this.prevProps,
+				{props} = this,
+				hasMetricsChanged = (
+					direction !== props.direction ||
+					((itemSize instanceof Object) ? (itemSize.minWidth !== props.itemSize.minWidth || itemSize.minHeight !== props.itemSize.minHeight) : itemSize !== props.itemSize) ||
+					overhang !== props.overhang ||
+					spacing !== props.spacing
+				);
+
+			this.hasDataSizeChanged = (dataSize !== props.dataSize);
+
+			if (hasMetricsChanged || this.props.clientSize && this.primary === null && this.secondary === null) {
+				this.calculateMetrics(props);
+				this.updateStatesAndBounds(props, true);
+				this.setContainerSize();
+			} else if (this.hasDataSizeChanged) {
+				this.updateStatesAndBounds(props, true);
+				this.setContainerSize();
+			} else if (rtl !== props.rtl) {
+				const {x, y} = this.getXY(this.scrollPosition, 0);
+
+				this.cc = [];
+				if (type === Native) {
+					this.scrollToPosition(x, y, props.rtl);
+				} else {
+					this.setScrollPosition(x, y, 0, 0, props.rtl);
+				}
+			}
+
 			if (primary) {
 				this.positionItems();
 			}
 
+			const {cc} = this;
```

Comment:
I could not replace `cc` with`this.cc` in `{itemsRenderer({cc, initItemContainerRef, primary})}` due to `unexpected reserved word this`.